core/src/main/scala/kafka/utils/Time.scala

```diff
@@ -44,6 +46,8 @@ trait Time {
 
   def nanoseconds: Long
 
+  def hiResClockMs: Long = TimeUnit.NANOSECONDS.toMillis(nanoseconds)
```

Comment:
It seems that for the Scala implementation, the mock time implementation should just return the same time unless `sleep` is called. For the Java one, we have an optional `autoTick` functionality, but we don't care about that here.